content/guides/docker-concepts/running-containers/sharing-local-files.md

```diff
@@ -0,0 +1,211 @@
+
+Docker offers two primary storage options for persisting data and sharing files between the host machine and containers: volumes and bind mounts.
+
+### Volume versus Bind Mounts
+
+If you want to ensure that data generated or modified inside the container persists even after the container stops running, you would opt for a volume. [Learn more](/guides/docker-concepts/running-containers/persisting-container-data/) about volumes and it's use cases.
+
+If you have specific files or directories on your host system that you want to directly share with your container, like configuration files or development code, then you would use a bind mount. It's like opening a direct portal between your host and container for sharing. Bind mounts are ideal for development environments where real-time file access and sharing between the host and container are crucial.
+
+### Sharing files between a host and container
+
+Both `-v` (or `--volume`) and `--mount` flags used with the `docker run` command let you share files or directories between your local machine (host) and a Docker container. However, there are some key differences in their behavior and usage.
+
+The` -v` flag is simpler and more convenient for basic volume or bind mount operations. If the host location doesn’t exist when using `-v` or `--volume`, a directory will be automatically created.
+
+Imagine you're a developer working on a project. You have a source directory on your development machine where your code resides. When you compile or build your code, the generated artifacts (compiled code, executables, images, etc.) are saved in a separate subdirectory within your source directory. Let's call this subdirectory  `/HOST/PATH`. Now you want these build artifacts to be accessible within a Docker container running your application. Additionally, you want the container to automatically access the latest build artifacts whenever you rebuild your code.
+
+Here’s a way how you use `docker run` to start a container using a bind mount and map it to the container file location.
+
+```console
+$ docker run -v /HOST/PATH:/CONTAINER/PATH -it nginx
+```
+
+The `--mount` offers more advanced features and granular control, making it suitable for complex mount scenarios or production deployments. If you use `--mount` to bind-mount a file or directory that doesn't yet exist on the Docker host, the `docker run` command doesn't automatically create it for you but generates an error.
+
+```console
+$ docker run --mount type=bind,source=/HOST/PATH,target=/CONTAINER/PATH,readonly nginx
+```
+
+> It is recommended to use the `--mount` syntax instead of `-v`. It provides better control over the mounting process and avoids potential issues with missing directories.
```

Comment:
```suggestion
> **Note**
>
> Docker recommends using the `--mount` syntax instead of `-v`. It provides better control over the mounting process and avoids potential issues with missing directories.
```